usbsdmux/sd_regs.py

```diff
@@ -0,0 +1,513 @@
+
+        sector_size = self.SECTOR_SIZE * self.WRITE_BL_LEN
+        wp_grp_size = self.WP_GRP_SIZE * sector_size
+
+        result["sector_size_bytes"] = {
+            "value": sector_size,
+            "unit": "bytes",
+            "name": "sector size",
+        }
+        result["wp_grp_size_bytes"] = {
+            "value": wp_grp_size,
+            "unit": "bytes",
+            "name": "write protect group size",
+        }
+
+        return result
+
+
+class CSD_10(CSD_Common):
+    VDD_MIN_ENUM = [0.5, 1, 5, 10, 25, 35, 60, 100]
+    VDD_MAX_ENUM = [1, 5, 10, 25, 35, 45, 80, 200]
+
+    FIELDS = CSD_Common.FIELDS.copy()
+    FIELDS["C_SIZE"] = {
+        "slice": (73, 62),
+        "name": "device size",
+        "convert": lambda v: (v + 1),
+    }
+    FIELDS["VDD_R_CURR_MIN"] = {
+        "slice": (61, 59),
+        "name": "max. read current @VDD min",
+        "enum": VDD_MIN_ENUM,
+        "unit": "mA",
+    }
+    FIELDS["VDD_R_CURR_MAX"] = {
+        "slice": (58, 56),
+        "name": "max. read current @VDD max",
+        "enum": VDD_MAX_ENUM,
+        "unit": "mA",
+    }
+    FIELDS["VDD_W_CURR_MIN"] = {
+        "slice": (55, 53),
+        "name": "max. write current @VDD min",
+        "enum": VDD_MIN_ENUM,
+        "unit": "mA",
+    }
+    FIELDS["VDD_W_CURR_MAX"] = {
+        "slice": (52, 50),
+        "name": "max. write current @VDD max",
+        "enum": VDD_MAX_ENUM,
+        "unit": "mA",
+    }
+    FIELDS["C_SIZE_MULT"] = {"slice": (49, 47), "name": "device size multiplier", "convert": lambda v: 2 ** (v + 2)}
+
+    def get_computed(self):
+        result = super().get_computed()
+
+        device_size = self.C_SIZE * self.C_SIZE_MULT * self.READ_BL_LEN
+
+        result["device_size_bytes"] = {
+            "value": device_size,
+            "unit": "bytes",
+            "name": "device size",
+        }
+
+        return result
+
+
+class CSD_20(CSD_Common):
+    FIELDS = CSD_Common.FIELDS.copy()
+    FIELDS["C_SIZE"] = {
+        "slice": (69, 48),
+        "name": "device size",
+        "convert": lambda v: (v + 1) * 512 * 1024,
+        "unit": "bytes",
+    }
+
+
+def decode_csd(raw_hex):
+    raw = int(raw_hex, 16)
+    val = bitslice(raw, 127, 126)
+    if val == 0:
+        return CSD_10(raw_hex)
+    elif val == 1:
+        return CSD_20(raw_hex)
+    else:
+        raise ValueError(f"unknown CSD version {val}")
+
+
+class CID(RegisterDecoder):
+    @staticmethod
+    def convert_prinable(value, size):
+        result = ""
+        for i in reversed(range(size)):
+            c = chr((value >> 8 * i) & 0xFF)
+            result += c if c.isprintable() else "."
+        return result
+
+    FIELDS = {
+        "MID": {
+            "slice": (127, 120),
+            "name": "Manufacturer ID",
+            "enum": {
+                0x02: "SanDisk",
+                0x03: "SanDisk SD",
+                0x74: "Transcend",
+                0x9F: "Kingston SD",
+            },
+        },
```

Comment:
You could also add `0x1b: "Samsung",` to the list (see https://github.com/linux-automation/usbsdmux/pull/63#discussion_r1431122989).